### ISSUE TITLE
How to run this model？

### ISSUE THREAD

**Issue by mymuli** [bbenligiray/rml-cnn#1]
Dear Sir:
I've read your paper and I'm very interested in your experiments... but I need some help from you.
1. How to store datasets and ResNet in folders?
2 .Can you give me some advice on training the model from scratch?

**Comment by bbenligiray** [comment]
I've updated the [readme](https://github.com/bbenligiray/rml-cnn/blob/master/README.md) for instructions about the datasets and the pretrained model.

The code allows you to train from scratch, for example `python main.py nus_wide random robust_warp 100 0` would train a randomly initialized model on NUS-WIDE with fully supervised rml-cnn loss. See [here](https://github.com/bbenligiray/rml-cnn/blob/d3f9f28d43862c63558a59b4cc2f63f5e8948e07/main.py#L142) for how these parameters work. You can add an `--optimize` parameter to do hyperparameter optimization.

**Comment by mymuli** [comment]
@bbenligiray  Thank you very much first!
But in the **params.py**  file, I am confused about these parameters，**[No_lr_steps、Lr_patience、No_random_start、No_opt_iters]** ，Can you explain them? 
At the same time, what's the difference between parameter  **imagenet**  and parameter **imagenet2**?
Thank you again！

**Comment by bbenligiray** [comment]
`no_lr_steps = 2` -> there will be two learning rate steps. So if we start training with a learning rate of 0.1, after `val_loss` stops improving, continue training with 0.1*0.1=0.01 learning rate (so the first step with 0.1 and the second step with 0.01 learning rate)
`lr_patience=10` -> stop training if `val_loss` doesn't improve for 10 epochs

`no_random_starts=8`
`no_opt_iters=15` this means first try 8 random hyperparameter sets between the given intervals, then try to converge to the best result with 15-8=7 additional tries. See [this](https://scikit-optimize.github.io/#skopt.gp_minimize).

So let's you started optimizing your hyperparameters with `--optimize` and ended up on the border (for example you were searching between 0.1 and 0.001 learning rate and ended up on 0.001). This means that the optimal hyperparameter was likely outside your initial interval (the first interval is named **imagenet**). Then you would set the second interval (**imagenet2**) between 0.001 and 0.00001 and run `main.py` with `--optimize --cont`. This parses the earlier training result between 0.1 and 0.001 to make a more educated estimation rather than starting from the scratch.

Basically it's a bit of over-engineering on my part. If you are not going to use the `--cont` parameter, you can ignore **imagenet2**.

**Comment by mymuli** [comment]
@bbenligiray  I was inspired，Thank you again！
